src/toil/jobStores/fileJobStore.py

```diff
@@ -215,13 +215,41 @@ def getSize(cls, url):
 
     @classmethod
     def _readFromUrl(cls, url, writable):
-        with open(cls._extractPathFromUrl(url), 'r') as f:
-            writable.write(f.read())
+        """Writes the contents of a file to a source (writes url to writable).
+
+        Uses a buffer of 1Gb if file is larger than 1Gb, else
+        read the entire file into memory for writing.
+        """
+        file_path = cls._extractPathFromUrl(url)
+        with open(file_path, 'r') as f:
+            if os.path.getsize(file_path) > 1073741824:
+                while True:
+                    buffer_size = 1073741824 # 1Gb RAM buffer
```

Comment:
use a single constant here, rather than two copies.  Assigning in the loop doesn't accomplish anything.

Reply:
Generally avoid constants in the code. This should be a configuration variable (e.g. environment variable). At a minimum, the constant should be in a variable that is used in the code.

Reply:
This is the kind of configuration variable that is very hard for a user to set right or know when to set.  Unless there is a really good reason to expose it,  it would be better not too.

I think it would be better to have a single largish buffer size that is just used for all files and not try to tune it on a per-file basis.   Maybe 256M?

but yes, one constant in the code.